pulp_ansible/app/galaxy/v3/views.py

```diff
@@ -858,14 +858,39 @@ def get_list_serializer(self, *args, **kwargs):
         kwargs.setdefault("context", self.get_serializer_context)
         return self.list_serializer_class(*args, **kwargs)
 
+    def get_list_queryset(self):
+        """
+        Derived from CollectionVersionRetrieveMixin.get_queryset
+        """
+        if getattr(self, "swagger_fake_view", False):
+            # drf_spectacular get filter from get_queryset().model
+            # and it fails when "path" is not on self.kwargs
+            return CollectionVersion.objects.none()
+        distro_content = self._distro_content
+
+        # Get all the CVs but limit the final fields fetched to just the PK,
+        # which is sufficient for what the cv list pagination and serializer
+        # require. If all fields are fetched, memory consumption increases
+        # indefinitely with repeated queries.
+        collections = (
+            CollectionVersion.objects.select_related("content_ptr__contentartifact")
+            .filter(
+                pk__in=distro_content, namespace=self.kwargs["namespace"], name=self.kwargs["name"]
+            )
+            .only(
+                "pk",
+            )
```

Comment:
You need to also include the fields you want from the `select_related`. Looking at the `print(collections.query)` on my machine the returned SQL is `SELECT "ansible_collectionversion"."content_ptr_id" FROM "ansible_collectionversion"`.